4917_Sprint/009 - Haoran/Demo/src/test/java/TestDemo.java

```diff
@@ -0,0 +1,60 @@
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.lang.ArithmeticException;
+
+
+public class TestDemo {
+    @Test
+
+    public void testAdd() {
+        assertEquals(Demo.calculate(2.0,3.0, "+"),5.0, 0.0);
+        assertEquals(Demo.calculate(-2.0,3.0, "+"),1.0, 0.0);
+        assertEquals(Demo.calculate(-2.0,-3.0, "+"),-5.0, 0.0);
+        assertEquals(Demo.calculate(0.0,0.0, "+"),0.0, 0.0);
+    }
+
+    @Test
+
+    public void testMinus() {
+        assertEquals(Demo.calculate(2.0,3.0, "-"),-1.0, 0.0);
+        assertEquals(Demo.calculate(-1.0,-3.0, "-"),2.0, 0.0);
+        assertEquals(Demo.calculate(12.0,3.0, "-"),9.0, 0.0);
+        assertEquals(Demo.calculate(0.0,0.0, "-"),0.0, 0.0);
+    }
+
+    @Test
+
+    public void testMultiply() {
+        assertEquals(Demo.calculate(2.0,3.0, "*"),6.0, 0.0);
+        assertEquals(Demo.calculate(-1.0,-3.0, "*"),3.0, 0.0);
+        assertEquals(Demo.calculate(12.0,3.0, "*"),36.0, 0.0);
+        assertEquals(Demo.calculate(0.0,0.0, "*"),0.0, 0.0);
+    }
+
+    @Test
+
+    public void testDivide() {
+        assertEquals(Demo.calculate(3.0,3.0, "/"),1.0, 0.0);
+        assertEquals(Demo.calculate(-3.0,-1.0, "/"),3.0, 0.0);
+        assertEquals(Demo.calculate(12.0,3.0, "/"),4.0, 0.0);
+
+    }
+
+
+    @Test
+
+    public void testArithmeticException(){
+        boolean test = false;
+        try {
+            Demo.calculate(1.0, 0.0, "/");
+        } catch(ArithmeticException e){
+            test = true;
```

Comment:
there is a better way to check the exception. Look at the options on the @Test annotation on line 46